sycl/include/sycl/detail/stl_type_traits.hpp

```diff
@@ -11,6 +11,7 @@
 #include <sycl/detail/defines_elementary.hpp>
 
 #include <iterator>
+#include <memory>
```

Comment:
@jsji, could you point to the declaration from memory header we use in sycl/include/sycl/detail/stl_type_traits.hpp. I don't see anything that needs this include.

> We met build failure in building sycl cts on Ubuntu 22 after https://github.com/intel/llvm/pull/10458.

We use Ubuntu 22 in our CI. We don't run SYCL-CTS, but I'm sure our end-to-end tests cover most of SYCL functionality. I suppose it might be related to GNU standard library version on your Ubuntu 22 system.